src/charts/scale.js

```diff
@@ -55,12 +55,16 @@ export default function Scale () {
   function buildXScale (_allKeys) {
     let xScale = null
     let domain = null
-    const markW =  hasBars(config.chartType) ? cache.chartWidth / _allKeys.length : 0
+    const isBarChart = hasBars(config.chartType)
+    const markW = 0 //  hasBars(config.chartType) ? cache.chartWidth / _allKeys.length : 0
 
     if (config.keyType === "time") {
       xScale = d3.scaleTime()
     } else if (config.keyType === "number") {
       xScale = d3.scaleLinear()
+    } else if (config.keyType === "string" && isBarChart) {
+      xScale = d3.scaleBand()
```

Comment:
I like the idea of using `scaleBand` for bar chart, which is more typical d3. Let's test the side-effects together and evaluate the alternative I will provide in another PR. Then we can decide which solution to adopt.

Reply:
I think it makes total sense to let D3 do the work when possible, rather than doing it manually oneself.

Reply:
I'm just worried about side-effects, for example with matching line `scalePoint` with bar `scaleBand`, when doing numerical with stacked, etc. We can try both and see which one as less side-effects, and revisit later for polishing. I want us to keep some time after 3.6 to polish the architecture and think about developer experience.